xskillscore/tests/test_probabilistic.py

```diff
@@ -129,205 +183,179 @@ def test_crps_quadrature_args(o_dask, f_prob_dask, keep_attrs):
+        # test for numerical identity of xs threshold and properscoring threshold
+        if keep_attrs:
+            expected = expected.assign_attrs(**actual.attrs)
+        assert_identical(actual, expected)
+    # test that returns chunks
+    assert_chunk(actual, chunk_bool)
+    # test that attributes are kept
+    assert_keep_attrs(actual, o, keep_attrs)
+    # test that input types equal output types
+    assign_type_input_output(actual, o)
 
 
-@pytest.mark.parametrize("keep_attrs", [True, False])
-def test_threshold_brier_score_dask_b_int(o_dask, f_prob_dask, keep_attrs):
-    threshold = 0
-    actual = threshold_brier_score(
-        o_dask, f_prob_dask, threshold, keep_attrs=keep_attrs
-    )
-    assert actual is not None
-    if keep_attrs:
-        assert actual.attrs == o_dask.attrs
-    else:
-        assert actual.attrs == {}
+@pytest.mark.parametrize("dim", DIMS)
+def test_threshold_brier_score_dim(o, f_prob, dim):
+    """Check that threshold_brier_score reduces only dim."""
+    actual = threshold_brier_score(o, f_prob, threshold=0.5, dim=dim)
+    assert_only_dim_reduced(dim, actual, o)
 
 
-@pytest.mark.parametrize("keep_attrs", [True, False])
-def test_threshold_brier_score_multiple_thresholds_list(o, f_prob, keep_attrs):
-    threshold = [0.1, 0.3, 0.5]
-    actual = threshold_brier_score(o, f_prob, threshold, keep_attrs=keep_attrs)
-    assert actual.chunks is None or actual.chunks == ()
-    if keep_attrs:
-        assert actual.attrs == o.attrs
-    else:
-        assert actual.attrs == {}
+@pytest.mark.parametrize(
+    "threshold",
+    [0, 0.5, [0.1, 0.3, 0.5], xr.DataArray([0.1, 0.3, 0.5], dims="threshold")],
+    ids=["int", "flat", "list", "xr.DataArray"],
+)
+def test_threshold_brier_score_threshold(o_dask, f_prob_dask, threshold):
+    """Test that threshold_brier_score accepts different kinds of thresholds."""
+    actual = threshold_brier_score(o_dask, f_prob_dask, threshold)
+    assert (actual.threshold == threshold).all()
+
+
+@pytest.mark.skip(reason="not implemented")
+def test_threshold_brier_score_threshold_dataset(o_dask, f_prob_dask):
+    """Test that threshold_brier_score accepts xr.Dataset thresholds."""
+    threshold = xr.DataArray([0.1, 0.3, 0.5], dims="threshold").to_dataset(name="var")
+    threshold["var2"] = xr.DataArray([0.2, 0.4, 0.6], dims="threshold")
+    o_dask = o_dask.to_dataset(name="var")
+    o_dask["var2"] = o_dask["var"] ** 2
+    f_prob_dask = f_prob_dask.to_dataset(name="var")
+    f_prob_dask["var2"] = f_prob_dask["var"] ** 2
+    actual = threshold_brier_score(o_dask, f_prob_dask, threshold)
+    assert actual
+    # test thresholds in each DataArray
+    # assert (actual.threshold == threshold).all()
 
 
-@pytest.mark.parametrize("keep_attrs", [True, False])
-def test_threshold_brier_score_multiple_thresholds_xr(o, f_prob, keep_attrs):
+def test_threshold_brier_score_dataset(o_dask, f_prob_dask):
+    """Test that threshold_brier_score accepts xr.Datasets."""
```

Comment:
Want to make a note you are testing dask arrays here?

Reply:
the test case multi-dim inputs is already covered above. removed